VolleyPaint/Assets/Scripts/Player/Shooting/PlayerShoot.cs

```diff
@@ -65,7 +70,30 @@ void Update()
 
         fireRateCountDown += Time.deltaTime;
 
-        if (Input.GetButton("Fire1") && fireRateCountDown >= fireRate)
+        if (isReloading)
+        {
+            reloadTimeElapsed += Time.deltaTime;
+
+            // Trigger for when to play reload sound
+            if (reloadTimeElapsed >= reloadCooldown / 2 && !reloadSoundPlayed)
+            {
+                reloadSoundPlayed = true;
+                MasterAudio.PlaySound3DAtVector3("Reload", transform.position);
+            }
+
+            // Manage animations for the gun
+            gunController.ReloadGunAnimation(reloadTimeElapsed, reloadCooldown);
+
+            // End reloading
+            if (reloadTimeElapsed >= reloadCooldown)
+            {
+                isReloading = false;
+                reloadSoundPlayed = false;
+                reloadTimeElapsed = 0.0f;
+                bulletsLeft = bulletLimit;         
+            }
+        }
+        if (!isReloading && Input.GetButton("Fire1") && fireRateCountDown >= fireRate)
```

Comment:
nit pick but u could do an else if here and remove the !isReloading

Reply:
or do something like this:

else {
   if(Input.GetButton("Fire1") && fireRateCountDown >= fireRate){
       ...
   }
   if(Input.GetKey(KeyCode.R) && bulletsLeft != bulletLimit){
      ...
   }
}